src/main/java/com/powsybl/openloadflow/sa/AcSecurityAnalysis.java

```diff
@@ -98,18 +113,25 @@ SecurityAnalysisReport runSync(String workingVariantId, SecurityAnalysisParamete
         return new SecurityAnalysisReport(result);
     }
 
-    List<LfNetwork> createNetworks(Set<Switch> allSwitchesToOpen, LfNetworkParameters networkParameters,
+    List<LfNetwork> createNetworks(Set<Switch> allSwitchesToOpen, Set<Switch> allSwitchesToClose, LfNetworkParameters networkParameters,
                                    Reporter saReporter) {
         List<LfNetwork> lfNetworks;
         String tmpVariantId = "olf-tmp-" + UUID.randomUUID();
+        String variantId = network.getVariantManager().getWorkingVariantId();
         network.getVariantManager().cloneVariant(network.getVariantManager().getWorkingVariantId(), tmpVariantId);
+        network.getVariantManager().setWorkingVariant(tmpVariantId);
         try {
             network.getSwitchStream().filter(sw -> sw.getVoltageLevel().getTopologyKind() == TopologyKind.NODE_BREAKER)
                     .forEach(sw -> sw.setRetained(false));
             allSwitchesToOpen.forEach(sw -> sw.setRetained(true));
+            allSwitchesToClose.forEach(sw -> {
+                sw.setRetained(true);
+                sw.setOpen(false); // in order to be present in the network.
```

Comment:
This is the ugly part of this PR. Not easy to do otherwise.

Reply:
:-) thanks! Note that there is for me a bug in the current code because we create a variant but we don't use it.